### ISSUE TITLE
False positive: assignment-from-no-return when decorator returns value

### ISSUE THREAD

**Issue by tuukkamustonen** [PyCQA/pylint#2414]
When a decorator returns something for a function call, it's `pylint` does not notice that and reports `assignment-from-no-return`.

With `test.py`:

```python
def return_1(method):
    def inner(*args, **kwargs):
        return 1
    return inner

@return_1
def where():
    pass

q = where()
print(q)
```

```shell
$ pylint test.py
...
fufu.py:10:0: E1111: Assigning to function call which doesn't return (assignment-from-no-return)
...
```

Something like this is used in [peewee](https://github.com/coleifer/peewee) (see https://github.com/coleifer/peewee/blob/master/peewee.py#L611-L617 and, for example, https://github.com/coleifer/peewee/blob/master/peewee.py#L1569-L1573).

`pylint --version`

```
pylint 2.1.1
astroid 2.0.4
Python 3.5.5 (default, May 29 2018, 10:15:24) 
[GCC 5.4.0 20160609]
```

**Comment by tuukkamustonen** [comment]
Oh, superb!
